internal/driver/glfw/window.go

```diff
@@ -240,21 +242,26 @@ func (w *window) SetIcon(icon fyne.Resource) {
 		return
 	}
 
-	if string(icon.Content()[:4]) == "<svg" {
-		fyne.LogError("Window icon does not support vector images", nil)
-		return
-	}
-
 	w.runOnMainWhenCreated(func() {
 		if w.icon == nil {
 			w.viewport.SetIcon(nil)
 			return
 		}
 
-		pix, _, err := image.Decode(bytes.NewReader(w.icon.Content()))
-		if err != nil {
-			fyne.LogError("Failed to decode image for window icon", err)
-			return
+		var (
```

Comment:
I think that `err` has the wrong scope here - it only applies to one branch of the `if`